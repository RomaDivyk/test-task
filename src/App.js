import ImageList from "./components/ImageList";

import "./App.css";

function App() {
  return (
    <div className="App">
      <h1>TEST APP</h1>
      <ImageList />
    </div>
  );
}

export default App;
