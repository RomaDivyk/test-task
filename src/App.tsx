import * as React from 'react';
import { Routes, Route } from 'react-router-dom';

import { ImageList, Form } from './components';

function App() {
  return (
    <div data-testid="appId">
      <h1>Test task</h1>
      <Routes>
        <Route path="/" element={<ImageList />} />
        <Route path="/2" element={<h1>TEST PAGE 2</h1>} />
        <Route path="/3" element={<h1>TEST PAGE 3</h1>} />
        <Route path="/auth" element={<Form />} />
      </Routes>
    </div>
  );
}

export default App;
