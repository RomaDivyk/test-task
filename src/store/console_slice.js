import { createSlice } from "@reduxjs/toolkit";

const consoleSlice = createSlice({
  name: "cons",
  initialState: {
    value: "Hello World!",
  },
  reducers: {
    showConsole: (state) => {
      console.log(state, "REDUX IS WORKING!!!!");
    },
  },
});

export const consoleAction = consoleSlice.actions;

export default consoleSlice.reducer;
