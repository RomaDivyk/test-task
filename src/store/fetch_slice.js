import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  data: [],
  isLoading: false,
};

const fetchSlice = createSlice({
  name: 'fetchData',
  initialState: initialState,
  reducers: {
    getDataFetch: (state) => {
      state.isLoading = true;
    },
    loadDataReduc: (state, action) => {
      state.data = action.payload;
    },
  },
});

export const { loadDataReduc, getDataFetch } = fetchSlice.actions;

export default fetchSlice.reducer;
