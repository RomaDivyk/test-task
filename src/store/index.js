import { configureStore } from "@reduxjs/toolkit";
import consoleSlice from "./console_slice";

const store = configureStore({
  reducer: {
    cons: consoleSlice,
  },
});

export default store;
