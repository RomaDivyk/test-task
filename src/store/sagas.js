import axios from "axios";
import { put, takeLatest } from "redux-saga/effects";

import { loadData } from "./fetch_slice";

const url = `https://demo1353770.mockable.io/images`;

function* getDataSaga() {
  try {
    const response = yield axios.get(url);
    console.log("HELLO SAGA");
    if (!response.ok) {
      throw new Error(`Something wrong in fetch data`);
    }

    yield put(loadData(response.data));
  } catch (error) {
    console.log(error);
  }
}

export function* watchGetData() {
  yield takeLatest("GET_DATA", getDataSaga);
}
