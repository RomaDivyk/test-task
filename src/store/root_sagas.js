import { all, fork } from "redux-saga/effects";
import { watchGetData } from "./sagas";

const rootSaga = function* () {
  yield all([fork(watchGetData)]);
};

export default rootSaga;
