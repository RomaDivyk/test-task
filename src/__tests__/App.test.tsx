import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../redux';

import App from '../App';
import { render, screen } from '@testing-library/react';

describe('testing App component', () => {
  it('should match a snapshot in App component', () => {
    const wrapper = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render App component with id', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
      { wrapper: BrowserRouter }
    );
    const element = screen.getByTestId('appId');
    expect(element).toBeInTheDocument();
  });
});
