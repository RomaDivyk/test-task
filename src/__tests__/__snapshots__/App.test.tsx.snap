// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App component should match a snapshot in App component 1`] = `
<div
  data-testid="appId"
>
  <h1>
    Test task
  </h1>
  <div
    className="sc-beySPh"
    data-testid="imageListId"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      data-testid="mediaCartId"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
      />
    </div>
    <div
      className="sc-dmyCSP"
    />
  </div>
</div>
`;
